streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -357,256 +274,197 @@ public static void verify(final String kafka, final Map<String, Set<Integer>> al
         }
 
         boolean success;
-        success = allData.equals(received);
+
+        final Map<String, HashSet<Number>> received =
+            events.get("echo")
+                  .entrySet()
+                  .stream()
+                  .map(entry -> mkEntry(entry.getKey(), new HashSet<>(entry.getValue())))
+                  .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        success = inputs.equals(received);
 
         if (success) {
             System.out.println("ALL-RECORDS-DELIVERED");
         } else {
             int missedCount = 0;
-            for (final Map.Entry<String, Set<Integer>> entry : allData.entrySet()) {
+            for (final Map.Entry<String, Set<Integer>> entry : inputs.entrySet()) {
                 missedCount += received.get(entry.getKey()).size();
             }
             System.out.println("missedRecords=" + missedCount);
         }
 
-        success &= verifyMin(min, allData, true);
-        success &= verifyMax(max, allData, true);
-        success &= verifyDif(dif, allData, true);
-        success &= verifySum(sum, allData, true);
-        success &= verifyCnt(cnt, allData, true);
-        success &= verifyAvg(avg, allData, true);
-        success &= verifyTAgg(tagg, allData, true);
+        // give it one more try if it's not already passing.
+        if (!verificationResult.passed()) {
+            verificationResult = verifyAll(inputs, events);
+        }
+        success &= verificationResult.passed();
```

Comment:
Should this be `|=` ?

Reply:
no, it's only a success if *all* the checks pass. So I think it's correct as an "and". Does that seem right?

Reply:
Ack. I guess I mixed up old code and new code -- also the comment `give it one more try if it's not already passing` is confusing. The interplay between `verificationResult.passed()` and `success` seems a little convoluted.